squad-compare-builds

```diff
@@ -146,16 +153,58 @@ def run():
 
     basefile_open = open(basefile_with_ending, 'r')
     otherfile_open = open(otherfile_with_ending, 'r')
-    difference = list(difflib.unified_diff(list(basefile_open.readlines()), list(otherfile_open.readlines()), fromfile=basefile_with_ending, tofile=otherfile_with_ending))
+    basefile_lines = basefile_open.readlines()
+    otherfile_lines = otherfile_open.readlines()
+    difference = list(difflib.unified_diff(basefile_lines, otherfile_lines, fromfile=basefile_with_ending, tofile=otherfile_with_ending))
     difference = ''.join(difference)
 
     filename = args.filename or f'diff-{basefile}-vs-{otherfile}.txt'
 
     with open(filename, 'w') as fp:
         fp.write(difference)
-    print(difference)
-    print(f"file created: {filename}")
 
+    tests = dict()
+    for line in basefile_lines:
+        test_name, test_result = line.split()
+        tests[test_name] = (test_result, None)
+    for line in otherfile_lines:
+        test_name, test_result = line.split()
+        if test_name in tests:
+            tests[test_name] = (tests[test_name][0], test_result)
+        else:
+            tests[test_name] = (None,test_result)
+
+    table_str = ""
+    first = True
+    for test_name, test_result in tests.items():
+        if test_result[0] != test_result[1]:
+            if first:
```

Comment:
I see you're using `first` just to add `\n` in between lines. 
A more elegant solution is to store each line in a list, then `table_str = '\n    '.join(lines)`